GUI.py

```diff
@@ -222,16 +232,30 @@ def startPlayback(self):
 
     def pauseAudio(self):
         '''Pause the currently playing audio'''
+        if self.audio.playing:
+            self.audio.paused = not self.audio.paused #toggle pause state
+
+            if self.audio.paused:
+                self.audio.pause()
+                #safely update button states using `after()`
+                self.after(0, lambda: self.pauseButton.configure(text="Resume"))
+                self.after(0, lambda: self.playButton.configure(state="disabled"))
+            else:
+                self.audio.resume() #resume audio if it was paused
```

Comment:
Remove unnecessary comments: Code line 244
The comment explaining the self.audio.resume() is redundant, as the function name already implies what’s happening. I suggest removing this comment to keep the code concise.